core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -331,7 +332,8 @@ private[spark] class MemoryStore(
     var unrollMemoryUsedByThisBlock = 0L
     // Underlying buffer for unrolling the block
     val redirectableStream = new RedirectableOutputStream
-    val bbos = new ChunkedByteBufferOutputStream(initialMemoryThreshold.toInt, allocator)
+    val bbos =
```

Comment:
Rather than import a whole other class for this, you probably want to explicitly check and throw an exception.

But, you're basically saying that this threshold can't be >= 4GB. Should that not just be checked earlier?